### DISCUSSION TITLE
Is it possible in app-vite to display static content before JS loads in SPA mode?

### DISCUSSION THREAD

**Discussion by gomander** [quasarframework/quasar#17647] [General - Components / Directives / etc] [answered]
In app-webpack v3, the HTML file has a `<div id="q-app">`, which we have populated with some static markup. Yes, I know that there's a warning against doing that, but hear me out.  
That content is then replaced once JS loads and the Vue app starts. This works really well for us, in that it displays some content while the app is loading, which can be a while on slow connections, and improves SEO for search engines that do not execute JS. We haven't seen any downside to doing things this way, and it does benefit us in at least these two ways.

We want to switch to app-vite v2, but unfortunately, we've not been able to find a way to do something similar in app-vite (or app-webpack v4), due to that div being replaced by a comment. Is there any way to accomplish something to this effect? This is very important for us because we have over 500 000 users every day, many of whom are in areas with slow internet connections, and SEO is very important to maintaining our search ranking.

**Comment by rstoenescu** [comment, marked as answer]
Hi, yes you can. Just write your own div, give it an id attribute and then in the JS code remove it from DOM
